### PR TITLE
Merging in features from week 7 - part one 

### PR DESCRIPTION
**Pull request by dcousette**
I'm merging in the following features from part one of week 7:  
- Built wrapper for Stripe API. 
- Charging credit card on registration. 
- Feature test for registration  

**I'm dealing with 2 failing specs, and still working to correct them.

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -9,11 +9,14 @@
...
 
   describe 'POST create' do
-    context 'with valid input' do
+    context 'with valid personal info and valid card' do
+      let(:charge) { double(:charge, successful?: true) }
       let(:alice) { Fabricate(:user) }
       let(:invitation) { Fabricate(:invitation, inviter: alice,
                               recipient_email: 'joe@example.com') }
 
+      before { StripeWrapper::Charge.should_receive(:create).and_return(charge) }
```
Comment (brandoncc) [inline spec/controllers/users_controller_spec.rb]: You should use `expect` syntax instead.

``` ruby
expect(StripeWrapper::Charge).to receive(:create).and_return(charge)
```

File: spec/features/charge_credit_card.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
+require 'spec_helper'
+
+feature 'Charging a credit card', { js: true } do
+  background { visit register_path }
+
+  scenario 'signup with valid user info and valid credit card' do
+    VCR.use_cassette('create_stripe_charge') do
+      fill_in 'Email Address', with: 'joesmith@gmail.com'
```
Comment (brandoncc) [inline spec/features/charge_credit_card.rb]: You can refactor all of these detail steps that are repeated into methods, like the solutions do.

File: spec/features/invite_user_features_spec.rb [hunk 3]
```diff
@@ -1,37 +1,44 @@
...
-  scenario "User successfully sends and accepts an invitation" do
-    john = Fabricate(:user)
-    sign_in(john)
-    click_on "Welcome, #{john.full_name}"
-    click_on "Invite a friend"
-    fill_in "Friend's Name", with: 'Joe Smith'
-    fill_in "Friend's Email Address", with: 'jsmith@gmail.com'
-    click_on "Send invitation"
+  scenario "User successfully sends and accepts an invitation", { js: true } do
+    VCR.use_cassette('create_stripe_charge') do
```
Comment (brandoncc) [inline spec/features/invite_user_features_spec.rb]: I suspect it is possible that your failing spec could come from the fact that you are sharing this cassette. I would let VCR handle the cassette naming itself.

File: spec/models/stripe_wrapper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            },
+          ).id
+
+          response = StripeWrapper::Charge.create(
+            amount: 999,
+            source: token,
+            description: 'a valid charge'
+          )
+
+          expect(response.successful?).to be_truthy
```
Comment (brandoncc) [inline spec/models/stripe_wrapper_spec.rb]: You can also do:

``` ruby
expect(response).to be_successful
```

File: spec/models/stripe_wrapper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            },
+          ).id
+
+          response = StripeWrapper::Charge.create(
+            amount: 999,
+            source: token,
+            description: 'an invalid charge'
+          )
+
+          expect(response.successful?).to be_falsey
```
Comment (brandoncc) [inline spec/models/stripe_wrapper_spec.rb]: Same here, except with `not_to`

**Comment by dcousette** [comment]
I've made changes based on suggestions. All tests are now passing as well. Merging to staging branch.

**Pull request merged**